ibis/expr/types/relations.py

```diff
@@ -2442,7 +2442,7 @@ def join(
 
         # semi/anti join only give access to the left table's fields, so
         # there's never overlap
-        if how in ("semi", "anti"):
+        if how in ("left_semi", "semi", "anti"):
```

Comment:
May want to update the comment above this line.

Reply:
`left_semi` and `semi` are the same thing and there's no `right_semi`, so the comment is correct if incomplete. I'll leave it as is for now.